golem/task/taskserver.py

```diff
@@ -495,8 +496,13 @@ def task_given(
             logger.error("Trying to assign a task, when it's already assigned")
             return False
 
-        self.task_computer.task_given(msg.compute_task_def)
-        if msg.want_to_compute_task.task_header.environment_prerequisites:
+        task_header: dt_tasks.TaskHeader = msg.want_to_compute_task.task_header
+
+        cpu_time_limit = task_helpers.calculate_max_usage(
+            task_header.subtask_budget, msg.price)

```

Comment:
We should use price from WTCT as price in TTC is subtask budget, rather than price per hour
```suggestion
            task_header.subtask_budget, msg.want_to_compute_task.price)
```